device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala

```diff
@@ -54,13 +54,15 @@ class GroupsResource(namespaceExtractor: Directive1[Namespace], deviceNamespaceA
                                 matching: Json,
                                 discarded: Json): Future[Int] = {
     val dbIO = for {
-      matchingGroups <- list(ns).map(
-                          _.filter{ info =>
-                            compare(disregard(info.systemInfo, discarded), matching).equals((matching, Json.Null))
-                          }.map(_.uuid))
-      res            <- DBIO.sequence(matchingGroups.map { device =>
-                          GroupMemberRepository.addOrUpdateGroupMember(groupId, device)
-                        })
+      allDevices <- list(ns)
+      matchingDevices = allDevices.collect {
+        case info if compare(disregard(removeIdNrs(info.systemInfo), discarded), matching)
+                       .equals((matching, Json.Null))
+          => info.uuid
+      }
+      res <- DBIO.sequence(matchingDevices.map { device =>
+                             GroupMemberRepository.addOrUpdateGroupMember(groupId, device)
+                           })
     } yield res
```

Comment:
Also this whole code should not be in this resource this is getting to complicated. This should all be extracted to a separate class that handles all this mess and can be tested separatly if needed